### PR TITLE
chore(python job): add JobOptions

### PR DESCRIPTION
**Pull request by nullndr**
## Changelog

This PR follows #1795, adding the typed dict `JobOptions` class for the python class `Job`.

In the previous, I forgot to add `total=False` to the `WorkerOptions` class, so I added it here. Sorry!

## Notes

Please tell me if it is better to move these typed dict classes in a subfolder like `types` or something, since these types will grown once the python library will add more things to match the node version

### CODE REVIEW

**Comment by manast** [comment]
> Please tell me if it is better to move these typed dict classes in a subfolder like `types` or something since these types will grow once the Python library will add more things to match the node version

In the TS version of the library, we have a directory "interfaces", I am not sure what is the standard naming for Python maybe it is "typing" if so then sure, it is a good way to organize the files as they will grow in number.

**Review by manast — APPROVED** [review]

File: python/bullmq/job.py [hunk 1]
```diff
@@ -13,6 +13,77 @@
 optsEncodeMap = {v: k for k, v in optsDecodeMap.items()}
 
 
+class KeepJosb(TypedDict, total=False):
```
Comment (ccollie) [inline python/bullmq/job.py:16]: Should be KeepJobs

Comment (nullndr) [reply python/bullmq/job.py:16]: Good catch

**Comment by nullndr** [comment]
@manast can we merge this? I'm going to open another PR to add the `QueueOptions` as well and I need this to update the `add` method signature

**Pull request merged**